### ISSUE TITLE
Implement getopts builtin

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#39]
Reported by timetoplatypus on lobste.rs

Also used in Alpine Linux `abuild` scripts

**Comment by timetoplatypus** [comment]
For clarity:

``` bash
function test() {
  while getopts ":h" opt; do
    case "$opt" in
      h)
        echo "Usage: test [PATH]" && return
      ;;
    esac
  done
}
```
Putting the above code in a file and running a simple `source` on that file will make the "test" function available to your shell. Invoking `test` will result in an error message
"Unexpected error in execvpe('getopts', ['getopts', ':h', 'opt'], ...): [Errno 2] No such file or directory"

**Comment by andychu** [comment]
Fixed wit the OSH 0.2 release.
